src/FactorySettings.php

```diff
@@ -0,0 +1,314 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\CircuitBreaker;
+
+use PrestaShop\CircuitBreaker\Contract\ClientInterface;
+use PrestaShop\CircuitBreaker\Contract\FactorySettingsInterface;
+use PrestaShop\CircuitBreaker\Contract\StorageInterface;
+use PrestaShop\CircuitBreaker\Contract\TransitionDispatcherInterface;
+
+/**
+ * Class FactorySettings is a simple implementation of FactorySettingsInterface, it is mainly
+ * a settings container and can be used with any Factory class.
+ */
+class FactorySettings implements FactorySettingsInterface
+{
+    /** @var int */
+    private $failures;
+
+    /** @var float */
+    private $timeout;
+
+    /** @var int */
+    private $threshold;
+
+    /** @var float */
+    private $strippedTimeout;
+
+    /** @var int */
+    private $strippedFailures;
+
+    /** @var StorageInterface|null */
+    private $storage;
+
+    /** @var TransitionDispatcherInterface|null */
+    private $dispatcher;
+
+    /** @var array */
+    private $clientOptions;
```

Comment:
```suggestion
    private $clientOptions = [];
```
And remove in constructor.

Reply:
Done